crates/agglayer-types/src/lib.rs

```diff
@@ -25,22 +32,183 @@ use pessimistic_proof::{
     LocalNetworkState, ProofError,
 };
 use serde::{Deserialize, Serialize};
-use unified_bridge::CommitmentVersion;
+use unified_bridge::{CommitmentVersion, LocalExitTree, LocalExitTreeError};
+
+// TODO: consider moving these newtype structs to interop some day, so that the
+// proofs could use eg. CertificateId and Height at least
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Default,
+    Eq,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    derive_more::Display,
+    derive_more::From,
+    serde::Deserialize,
+    serde::Serialize,
+)]
+#[cfg_attr(feature = "testutils", derive(arbitrary::Arbitrary))]
+#[serde(transparent)]
+pub struct EpochNumber(u64);
+
+impl EpochNumber {
+    pub const ZERO: EpochNumber = EpochNumber(0);
+    pub const ONE: EpochNumber = EpochNumber(1);
+
+    pub const fn new(epoch: u64) -> EpochNumber {
+        EpochNumber(epoch)
+    }
+
+    pub const fn next(&self) -> EpochNumber {
```

Comment:
Elaborating on the discussion below on the potential misunderstanding whether given methods updates the value in place or returns the new value. A useful thing to do in this case is to require the user to do something with the result.

```suggestion
    #[must_use = "The value of the next epoch is returned but not used"]
    pub const fn next(&self) -> EpochNumber {
```